frontend/catalyst/api_extensions/quantum_passes.py

```diff
@@ -0,0 +1,92 @@
+# Copyright 2024 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+This module contains public API functions that provide control for the 
+user to input what MLIR compiler passes to run. 
+
+Currently, each pass has its own user-facing decorator. In the future, 
+a unified user interface for all the passes is necessary. 
+
+Note that the pass pipeline table does not need to modify the qnode in 
+any way. Its only purpose it to mark down the passes the user wants to 
+run on each qnode, and then send this information to
+frontend/catalyst/compiler.py to handle the actual running of the passes. 
+"""
+
+import pennylane as qml
+
+
+## PASS PIPELINE TABLE ##
+class QUANTUM_PASSES_TABLE:
+    """
+    A class that records the active compilation passes on a qnode.
+    Note that the ordering of passes of course matters.
+    """
+
+    def __init__(self):
+        self._table = {}
+
+    def __repr__(self):
```

Comment:
Is this really necessary?

Reply:
This isn't necessary. It's a convenient utility for printing but I guess there isn't any real point leaving this in. I have removed it.